draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -1084,6 +1094,23 @@ However, if the transparency service's interface becomes unavailable, monitors w
 
 {{trust-anchor-negotiation}} and {{tls-trust-anchors-extension}} involve the relying party sending a list of TrustAnchor values to aid the subscriber in selecting certificates. A sufficiently large list may be impractical to fit in a ClientHello and require alternate negotiation mechanisms or a different PKI structure. To reduce overhead, `issuer_id` values SHOULD be short, no more than eight bytes long.
 
+## Bitflips
+
+Bitflips happen at scale; indeed bitflips invalidated Certificate Transparency logs: see {{NESSIE2024BITFLIP}} and {{YETI2022BITFLIP}}.
+
+Where security allows it, the present system has been designed to be lenient with bitflips. For example:
+
+1. A signature that does not verify is not considered an misissuance, but rather an availability issue. This allows for bitflips in the signature itself and the signed message.
+2. An (abridged) assertion published via the HTTP interface, is ignored unless it's included in a valid signed window.
+3. A bitflip in a MerkleTreeProofSHA256 can be safely ignored, as any BikeshedCertificate that include it will fail to verify.
+
+There are several places where bitflips cannot be tolerated. For instance, misissuances include (but are not limited to)
+
+1. A bitflip in an assertion, which is included (with bitflip) in a valid signed window.
+2. A valid signed window, for which the CA cannot *eventually* produce assertions that lead to the included tree heads.
+
+CAs should take care to catch these bitflips. For instance, a CA should double-check the computation of tree heads before signing them.
```

Comment:
Given CT's ambiguity here left us in a bad spot wrt bit flips, could we refer to a strong practice or just prescribe specific steps here? e.g. Sign --> verify --> Publish or Sign1 --> Sign2 --> Compare --> Publish. Catching these before the signatures get into the wild really does reduce it to a pure availability problem.

Reply:
I'm happy to add a list of suggested countermeasures. However, I don't want to seem exhaustive.